packages/@lwc/engine-server/src/renderer.ts

```diff
@@ -36,6 +38,30 @@ function createElement(name: string, namespace?: string): HostElement {
     };
 }
 
+const registry: Record<string, CustomElementConstructor> = create(null);
+const reserveRegistry: WeakMap<CustomElementConstructor, string> = new WeakMap();
+
+function registerCustomElement(name: string, ctor: CustomElementConstructor) {
+    if (name !== StringToLowerCase.call(name) || registry[name]) {
+        throw new TypeError(`Invalid Registration`);
+    }
+    registry[name] = ctor;
+    reserveRegistry.set(ctor, name);
+}
+
+class HTMLElement {
+    constructor() {
+        const { constructor } = this;
+        const name = reserveRegistry.get(constructor as CustomElementConstructor);
+        if (!name) {
+            throw new TypeError(`Invalid Construction`);
+        }
+        const elm = createElement(name);
+        elm.constructor = this.constructor; // restoring constructor into the fake instance
+        return elm;
```

Comment:
the trick is to replace the element's constructor so it can be used by the subclassing to define the update static hook